pkg/labels/lister_test.go

```
@@ -0,0 +1,134 @@
+	if err := lister.ListTargets(context.Background(), &out); err != nil {
+		t.Fatalf("ListTargets() error = %v", err)
+	}
+	if out.Len() == 0 {
+		t.Fatal("expected non-empty output from bundled rulesets")
+	}
+}
+
+func TestNew_nilLoggerUsesDiscard(t *testing.T) {
+	lister := New(Config{})
+	// Must not panic when logging during list operations.
+	lister.cfg.Log.Info("init")
```

golangci-lint (2.12.2)

[error] SA5011: possible nil pointer dereference

(staticcheck)